### PR TITLE
feat:add the ant movement logic

### PR DESCRIPTION
**Pull request by WycliffeAlphus**
This PR introduces the Mover implementation for coordinating simultaneous ant movements . The movement logic respects the rules for StartRoom and EndRoom, ensuring they are not marked as occupied during movement steps.

### Key Changes
-  New Mover Struct:
     -  Manages ant movements with reference to the colony's structure.
     - Ensures adherence to rules for StartRoom and EndRoom.

- ExecuteMovements Function:
     - Returns a series of steps containing valid simultaneous movements of ants.

- executeNextStep Function:
     - Handles one turn of ant movements.
     -  Skips occupancy checks for StartRoom and EndRoom.
     -  Updates room occupancy dynamically for intermediate rooms.

-  Integration with Colony:
     -  StartRoom and EndRoom logic is integrated using the Colony reference.

### How It Works
- Initialization
     - Create a Mover instance with ants and the colony reference:
          - mover := NewMover(colony.Ants, colony)
          - movements := mover.ExecuteMovements()

### Movement Logic

Ants move to valid adjacent rooms unless the room is already occupied (excluding StartRoom and EndRoom).

Updates are logged in steps for clarity and output purposes.

### Testing
Tests are not included:
 Add comprehensive unit tests for edge cases (e.g., bottlenecks, cyclic paths).

### CODE REVIEW

**Review by rayjonesjay — APPROVED** [review]

File: controllers/mover.go [hunk 1]
```diff
@@ -1 +1,100 @@
...
+}
+
+// Mover coordinates simultaneous ant movements
+type Mover struct {
+	ants   []*models.Ant
+	colony *models.Colony // Reference to the Colony for start and end room
+}
+
+// NewMover creates a new Mover instance
+func NewMover(ants []*models.Ant, colony *models.Colony) *Mover {
```
Comment (josie-opondo) [inline controllers/mover.go:20]: IMO, the constructor should receive the colony and paths([][]*Room) since we'll be moving ants after we find paths

Comment (WycliffeAlphus) [reply controllers/mover.go:20]: The paths are part of the Ant struct, therefore they can be accessed through the pointer to models.Ant

Comment (josie-opondo) [reply controllers/mover.go:20]: Okay, understood

**Review by josie-opondo — APPROVED** [review]
These changes are acceptable and can be merged
